### PR TITLE
Play the coin sound effect when buying items from shop.

### PR DESCRIPTION
**Pull request by collett8192**
When selling the sound effect is also played (and shop switch to buy back tab as well) but the selling logic code is not there so...

### CODE REVIEW

File: src/world/Actor/PlayerEvent.cpp [hunk 1]
```diff
@@ -145,6 +145,8 @@ void Sapphire::Entity::Player::playGilShop( uint32_t eventId, uint32_t flags,
   openGilShopPacket->data().eventId = eventId;
   openGilShopPacket->data().sceneFlags = flags;
   openGilShopPacket->data().actorId = getId();
+  openGilShopPacket->data().params[ 0 ] = 1;
```
Comment (NotAdam) [inline src/world/Actor/PlayerEvent.cpp]: did I get the packet wrong? there's a field before this for the arg count

Comment (collett8192) [reply src/world/Actor/PlayerEvent.cpp]: this one is tricky. we only need the next param but this one cannot be 0. on retail it's 0x721A64A2 not looking like a count at all, and the count field we have is always 0

Comment (collett8192) [reply src/world/Actor/PlayerEvent.cpp]: ok the packet is wrong, see the latest update. I wonder why my wrong packet was working lol

Comment (NotAdam) [reply src/world/Actor/PlayerEvent.cpp]: yeah okay, maybe there's something else going on there, will need to double check the packet but this is fine for now

**Pull request merged**